bundles/org.eclipse.ui.ide/extensions/org/eclipse/ui/actions/WorkspaceAction.java

```diff
@@ -369,6 +369,9 @@ protected boolean updateSelection(IStructuredSelection selection) {
 		if (!super.updateSelection(selection) || selection.isEmpty()) {
 			return false;
 		}
+		if (getSelectedResources().size() > 1) {
```

Comment:
This is strange, what if all resources are inaccessible?

I propose the following:

```
		if (getSelectedResources().stream().allMatch(r -> !r.isAccessible())) {
			return false;
		}
```

Reply:
@sratz , If all resources are inaccessible, the 'right-click refresh' is not yet enabled.